java/vector/src/main/java/org/apache/arrow/vector/util/ValueVectorUtility.java

```diff
@@ -82,4 +89,43 @@ public static String getToString(ValueVector vector, int start, int end) {
 
     return sb.toString();
   }
+
+  /**
+   * Validate field vector.
+   */
+  public static void validate(FieldVector vector) {
+    Preconditions.checkNotNull(vector);
+
+    ArrowType arrowType = vector.getField().getType();
+    int typeBufferCount = TypeLayout.getTypeBufferCount(arrowType);
+    TypeLayout typeLayout = TypeLayout.getTypeLayout(arrowType);
+
+    if (vector.getValueCount() < 0) {
+      throw new RuntimeException("vector valueCount is negative");
```

Comment:
I think IllegalArgumentException is better?

Reply:
thanks, accepted.